server/reddit.js

```diff
@@ -31,33 +45,51 @@ redditApi.interceptors.response.use((response) => {
 
 const userAgent = 'node:com.vexillologycontests:v0.1.0';
 
-const snoowrap = new Snoowrap({
-  userAgent,
-  clientId: REDDIT_CLIENT_ID,
-  clientSecret: REDDIT_CLIENT_SECRET,
-  username: 'heshammourad',
-  password: REDDIT_PASSWORD,
-});
+let snoowrap;
+
+if (!IS_DEV) {
+  snoowrap = new Snoowrap({
+    userAgent,
+    clientId: REDDIT_CLIENT_ID,
+    clientSecret: REDDIT_CLIENT_SECRET,
+    username: REDDIT_USERNAME,
+    password: REDDIT_PASSWORD,
+  });
+}
 
+/**
+ * @param {object} [auth] user auth
+ * @returns auth ? user-populated snoowrap : default snoowrap
+ */
 const getSnoowrap = (auth = {}) => {
-  const accessToken = auth.accessToken || auth.accesstoken || auth.access_token;
-  const refreshToken = auth.refreshToken || auth.refreshtoken || auth.refresh_token;
-  if (!accessToken || !refreshToken) {
-    logger.debug('No auth tokens provided, returning default snoowrap');
-    return snoowrap;
+  const accessToken = auth.accessToken || auth.accesstoken || auth.access_token
+    || WEB_APP_ACCESS_TOKEN;
+  const refreshToken = auth.refreshToken || auth.refreshtoken || auth.refresh_token
+    || WEB_APP_REFRESH_TOKEN;
+
+  if (IS_DEV || (accessToken && refreshToken)) {
+    return new Snoowrap({
+      userAgent,
+      clientId: WEB_APP_CLIENT_ID,
+      clientSecret: WEB_APP_CLIENT_SECRET,
+      refreshToken,
+      accessToken,
+    });
   }
 
-  return new Snoowrap({
-    userAgent,
-    clientId: WEB_APP_CLIENT_ID,
-    clientSecret: WEB_APP_CLIENT_SECRET,
-    refreshToken,
-    accessToken,
-  });
+  logger.debug('No auth tokens provided, returning default snoowrap');
+  return snoowrap;
 };
 
+/**
+ * Fetch contest from database (or legacy reddit posts)
```

Comment:
This file does not interact with the database. It only reads data from Reddit.

Reply:
Edited